osu.Game/Beatmaps/Beatmap.cs

```diff
@@ -74,6 +75,31 @@ public double GetMostCommonBeatLength()
             return mostCommon.beatLength;
         }
 
+        public int ClosestSnapTime(double time, int beatDivisor, double? referenceTime = null)
+        {
+            var timingPoint = ControlPointInfo.TimingPointAt(referenceTime ?? time);
+            var beatLength = timingPoint.BeatLength / beatDivisor;
+            var beatLengths = (int)Math.Round((time - timingPoint.Time) / beatLength, MidpointRounding.AwayFromZero);
+
+            // Casting to int matches stable.
```

Comment:
I'm not sure we want this. @smoogipoo opinions? Until now we have been using doubles in the editor and this will force all hitobjects to be snapped to `int` time values.

Reply:
I agree, should be using doubles.